deepmd/utils/data.py

```diff
@@ -456,13 +468,17 @@ def _load_set(self, set_name: DPPath) :
                 data[kk] = np.sum(np.reshape(tmp_in, [nframes, self.natoms, ndof]), axis = 1)
 
         if self.mixed_type:
-            type_path = set_name / "real_atom_types.npy"
-            real_type = type_path.load_numpy().astype(np.int32).reshape([nframes, -1])
+            real_type = self.atom_type_mix.reshape([nframes, self.natoms])
             data['type'] = real_type
             natoms = data['type'].shape[1]
+            # nframes x ntypes
+            atom_type_nums = np.array([(real_type == i).sum(axis=-1) for i in range(self.get_ntypes())],
+                                      dtype=np.int32).T
```

Comment:
For a more elegant code I think you can try [`numpy.bincount`](https://numpy.org/doc/stable/reference/generated/numpy.bincount.html):
```py
atom_type_nums = np.bincount(real_type, minlength=self.get_ntypes()).T
```

Reply:
I tried np.bincount, but it does not support 2-dim array here (nframes x natoms), unless you split in the nframes dim, which is more ugly than 'for' in ntypes dim here.